### PR TITLE
[api] Move `Vars` and `Term` out of API.

### PR DESCRIPTION
**Pull request by ejgallego**
It looks like most of the API here should be public. Note the strong
deprecation in `Term` due the introduction of `Constr` a few years ago.

c.f. issue #6008

### CODE REVIEW

File: API/API.mli [hunk 1]
```diff
@@ -20,7 +20,7 @@
    See below in the file for their concrete position.
 *)
 
-(************************************************************************)
+(***************************************************************b*********)
```
Comment (gares) [inline API/API.mli]: what is this ;-)

File: API/API.mli [hunk 2]
```diff
@@ -31,992 +31,64 @@ end
...
-  val strip_lam : constr -> constr
-  val strip_prod_assum : types -> types
-
-  val decompose_lam_assum : constr -> Context.Rel.t * constr
-  val destFix : constr -> fixpoint
-
-  val compare_constr : (constr -> constr -> bool) -> constr -> constr -> bool
-end
+(* these constraints are workaround in order to make the aliases in
+   Term, etc... work; should go away once these modeles are ported to
```
Comment (gares) [inline API/API.mli]: modules, not modeles

**Comment by herbelin** [comment]
@ejgallego: You're asking for feedback, so, I shall try to give some, but, to be honest, I'm a bit lost in all the discussions. So, for the kernel, is the idea that it is already well-designed enough and that there is not much filtering to do?

**Comment by ejgallego** [comment]
> @ejgallego: You're asking for feedback, so, I shall try to give some, but, to be honest, I'm a bit lost in all the discussions. So, for the kernel, is the idea that it is already well-designed enough and that there is not much filtering to do?

Thanks @herbelin , I am not expert in the kernel so hard to say what should the "kernel API" be.

**Comment by ejgallego** [comment]
No internal modules here.

**Pull request closed without merging**